titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/DefaultV3JobOperations.java

```diff
@@ -152,21 +153,22 @@ public void enterActiveMode() {
         this.transactionLoggerSubscription = JobTransactionLogger.logEvents(reconciliationFramework);
 
         // Remove finished jobs from the reconciliation framework.
-        this.reconcilerEventSubscription = reconciliationFramework.events().subscribe(
-                event -> {
-                    if (event instanceof JobModelUpdateReconcilerEvent) {
-                        JobModelUpdateReconcilerEvent jobUpdateEvent = (JobModelUpdateReconcilerEvent) event;
-                        handleJobCompletedEvent(jobUpdateEvent.getChangedEntityHolder());
-                    }
-                },
-                e -> logger.error("Event stream terminated with an error", e),
-                () -> logger.info("Event stream completed")
-        );
+        this.reconcilerEventSubscription = titusRuntime.persistentStream(reconciliationFramework.events().onBackpressureBuffer())
```

Comment:
This creates unbounded buffer. I think we should do on back-pressure drop, and do full reconciliation at subscription time.

Reply:
changed